sphinx/builders/html/__init__.py

```diff
@@ -1215,18 +1215,14 @@ def setup_resource_paths(app: Sphinx, pagename: str, templatename: str,
     pathto = context.get('pathto')
 
     # favicon_url
-    favicon = context.get('favicon')
+    favicon = context.get('favicon_url')
     if favicon and not isurl(favicon):
         context['favicon_url'] = pathto('_static/' + favicon, resource=True)
```

Comment:
It's better to call this variable as `favicon_url`.
```suggestion
    favicon_url = context.get('favicon_url')
    if favicon_url and not isurl(favicon_url):
        context['favicon_url'] = pathto('_static/' + favicon_url, resource=True)
```